src/models/schema.ts

```diff
@@ -40,23 +40,14 @@ export class Schema extends Data implements IResourceBase {
   type!: SchemaType;
   // added separately from the response data, used for follow-on API calls
   schemaRegistryId!: Enforced<string>;
-  environmentId!: Enforced<string> | undefined;
+  environmentId!: EnvironmentId | undefined;
 
   /** Is this the highest version bound to this subject? */
   isHighestVersion!: Enforced<boolean>;
 
   /** Returns true if this schema subject corresponds to the topic name per TopicNameStrategy or TopicRecordNameStrategy*/
   matchesTopicName(topicName: string): boolean {
-    if (this.subject.endsWith("-key")) {
-      // TopicNameStrategy key schema
-      return this.subject === `${topicName}-key`;
-    } else if (this.subject.endsWith("-value")) {
-      // TopicNameStrategy value schema
-      return this.subject === `${topicName}-value`;
-    } else {
-      // only other possibility is a matching TopicRecordNameStrategy (value) schema
-      return this.subject.startsWith(`${topicName}-`);
-    }
+    return subjectMatchesTopicName(this.subject, topicName);
```

Comment:
Extracted the implementation here to a standalone function and made to take the subject name, not an entire `Schema` object.